api/org/commcare/api/persistence/SqlHelper.java

```diff
@@ -0,0 +1,246 @@
+package org.commcare.api.persistence;
+
+import org.commcare.modern.util.Pair;
+import org.commcare.modern.database.*;
+import org.javarosa.core.services.Logger;
+import org.javarosa.core.services.storage.Persistable;
+
+import java.io.ByteArrayInputStream;
+import java.sql.Blob;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * Set of Sql utility methods for clients running modern, non-Android Java (where prepared
+ * statements in the place of cursors)
+ *
+ * All methods that return a ResultSet expect a PreparedStatement as an argument and the caller
+ * is responsible for closing this statement when its finished with it.
+ *
+ * Created by wpride1 on 8/11/15.
+ */
+public class SqlHelper {
+    public static void dropTable(Connection c, String storageKey) {
+        String sqlStatement = "DROP TABLE IF EXISTS " + storageKey;
+        PreparedStatement preparedStatement = null;
+        try {
+            preparedStatement = c.prepareStatement(sqlStatement);
+            preparedStatement.execute();
+        } catch (SQLException e) {
+            Logger.log("E", "Could not drop table: " + e.getMessage());
+        } finally {
+            if (preparedStatement != null) {
+                try {
+                    preparedStatement.close();
+                } catch (SQLException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+    }
+
+    public static void createTable(Connection c, String storageKey, Persistable p) {
+        String sqlStatement = DatabaseHelper.getTableCreateString(storageKey, p);
+        PreparedStatement preparedStatement = null;
+        try {
+            preparedStatement = c.prepareStatement(sqlStatement);
+            preparedStatement.execute();
+        } catch (SQLException e) {
+            //fail silently, expected sometimes
+        } finally {
+            if (preparedStatement != null) {
+                try {
+                    preparedStatement.close();
+                } catch (SQLException e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+    }
+
+    public static PreparedStatement prepareIdSelectStatement(Connection c,
+                                                             String storageKey,
+                                                             int id) {
+        try {
+            PreparedStatement preparedStatement =
+                    c.prepareStatement("SELECT * FROM " + storageKey + " WHERE "
+                            + DatabaseHelper.ID_COL + " = ?;");
+            preparedStatement.setInt(1, id);
+            return preparedStatement;
+
+        } catch (SQLException e) {
+            e.printStackTrace();
```

Comment:
This doesn't seem right to me. We should be rethrowing as a runtime if we aren't going to process, shoudln't we?